File: Java/Medium/CompileError/most-profit-assigning-work_3.java
class Solution {
    public int maxProfitAssignment(int[] difficulty, int[] profit, int[] worker) {
        
        int n = difficulty.length;
        int[][] jobs = new int[n][2];
        for (int i = ; i < n; i++) {
            jobs[i][0] = difficulty[i];
            jobs[i][1] = profit[i];
        }
        Arrays.sort(jobs, (a, b) -> a[0] - b[0]);
        Arrays.sort(worker);
        int res = 0, i = 0
        for (int i = 0; i < worker.length; i++) {
            while (j < n && jobs[j][0] <= worker[i]) {
                max = Math.max(max, jobs[j][1]);
                j++;
            }
            res += max;
        }
        return res;
    }
}

File: C++/Medium/Accepted/most-profit-assigning-work_3.cpp
class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        
        int n = difficulty.size();
        vector<pair<int, int>> jobs;
        for (int i = 0; i < n; ++i) {
            jobs.emplace_back(difficulty[i], profit[i]);
        }
        sort(jobs.begin(), jobs.end());
        sort(worker.begin(), worker.end());
        int ans = 0, best = 0, i = 0;
        for (int ability: worker) {
            while (i < n && ability >= jobs[i].first) {
                best = max(best, jobs[i++].second);
            }
            ans += best;
        }
        return ans;
    }
};